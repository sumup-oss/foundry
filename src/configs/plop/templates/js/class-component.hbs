import React, { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * Describe {{ name }} here.
 */
export default class {{ name }} extends Component {
  static propTypes = {
    /**
    * A consice description of the example prop.
    */
    example: PropTypes.string
  };

  static defaultProps = {};

  render() {
    return null;
  }
}
