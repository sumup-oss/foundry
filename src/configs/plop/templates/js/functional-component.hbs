import React from 'react';
import PropTypes from 'prop-types';

/**
 * Describe {{ name }} here.
 */
const {{ name }} = (props) => {
  return null;
};

{{ name }}.propTypes = {
  /**
   * A consice description of the example prop.
   */
  example: PropTypes.string
};

/**
 * @component
 */
export default {{ name }};
