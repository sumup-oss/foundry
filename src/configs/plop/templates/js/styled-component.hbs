import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

const baseStyles = ({ theme }) => css`
  label: {{ dashCase name }};
`;

/**
 * Describe {{ name }} here.
 */
const {{ name }} = styled('element')(baseStyles);

{{ name }}.propTypes = {
  /**
   * A concise description of the example prop.
   */
  example: PropTypes.string
};

/**
 * @component
 */
export default {{ name }};
