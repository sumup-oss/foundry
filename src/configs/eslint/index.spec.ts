/**
 * Copyright 2025, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { describe, expect, it } from 'vitest';

import {
  Environment,
  Framework,
  Language,
  Plugin,
} from '../../types/shared.js';

import { files } from './index.js';

describe('eslint', () => {
  describe('files', () => {
    it('should match the snapshot for basic options', () => {
      const options = {};
      const actual = files(options);
      expect(actual).toMatchSnapshot();
    });

    it('should match the snapshot for complete options', () => {
      const options = {
        configDir: '.',
        language: Language.TypeScript,
        environments: [Environment.Browser, Environment.Node],
        frameworks: [Framework.React],
        plugins: [Plugin.Jest, Plugin.Storybook],
        openSource: true,
      };
      const actual = files(options);
      expect(actual).toMatchSnapshot();
    });
  });
});
