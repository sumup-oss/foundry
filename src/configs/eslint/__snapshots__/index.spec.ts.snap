// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`eslint > files > should match the snapshot for basic options 1`] = `
[
  {
    "content": "import { configs, defineConfig } from '@sumup-oss/foundry/eslint';


export default defineConfig([
  configs.ignores, configs.javascript
]);",
    "name": "eslint.config.mjs",
  },
]
`;

exports[`eslint > files > should match the snapshot for complete options 1`] = `
[
  {
    "content": "import { configs, defineConfig } from '@sumup-oss/foundry/eslint';
import react from 'eslint-plugin-react'

export default defineConfig([
  configs.ignores, configs.javascript, configs.typescript, configs.browser, configs.node, configs.openSource, { extends: [ react.configs.recommended, react.configs['jsx-runtime'], configs.react], plugins: { react }}, configs.tests, configs.stories
]);",
    "name": "eslint.config.mjs",
  },
]
`;
